FModel/ViewModels/ModelViewerViewModel.cs

```diff
@@ -395,20 +395,36 @@ private void PushLod(CMeshSection[] sections, CMeshVertex[] verts, FRawStaticInd
                         }
                         case FGame.ShooterGame:
                         {
-                            // Valorant's Specular Texture Channels
-                            // R Metallic
-                            // G Specular
-                            // B Roughness
-                            unsafe
+                            var packedPBRType = specular.Name[(specular.Name.LastIndexOf('_') + 1)..];
+                            switch (packedPBRType)
                             {
-                                var offset = 0;
-                                fixed (byte* d = data)
-                                    for (var i = 0; i < mip.SizeX * mip.SizeY; i++)
+                                case "MRAE": // R: Metallic, G: AO (0-127) & Emissive (128-255), B: Roughness   (Character PBR)
+                                    unsafe
                                     {
-                                        (d[offset], d[offset+2]) = (d[offset+2], d[offset]); // swap R and B
-                                        (d[offset], d[offset+1]) = (d[offset+1], d[offset]); // swap B and G
-                                        offset += 4;
+                                        var offset = 0;
+                                        fixed (byte* d = data)
+                                            for (var i = 0; i < mip.SizeX * mip.SizeY; i++)
+                                            {
+                                                (d[offset], d[offset+2]) = (d[offset+2], d[offset]); // swap R and B
+                                                (d[offset], d[offset+1]) = (d[offset+1], d[offset]); // swap R and G
+                                                offset += 4;
+                                            }
+                                    }
+                                    break;
+                                case "MRAS": // R: Metallic, B: Roughness, B: AO, A: Specular   (Legacy PBR)
+                                case "MRA":  // R: Metallic, B: Roughness, B: AO                (Environment PBR)
+                                case "MRS":  // R: Metallic, G: Roughness, B: Specular          (Weapon PBR)
+                                    unsafe
+                                    {
+                                        var offset = 0;
+                                        fixed (byte* d = data)
```

Comment:
Also here.

Reply:
consistency, Officer, if I add brackets in my part but not to Asval's then it will make no sense :)

Reply:
https://github.com/iAmAsval/FModel/pull/215/commits/dce3d1fa17a2c5aafbd8ff87d28a2edd1f404ee9